hub/core/compression/webp.py

```diff
@@ -0,0 +1,84 @@
+from io import BytesIO
+import numcodecs  # type: ignore
+import numpy as np
+from PIL import Image, ImageOps  # type: ignore
+
+from hub.core.compression import BaseImgCodec
+
+
+class WEBP(BaseImgCodec):
+    def __init__(self, **kwargs):
+        """
+        Initialize WebP compressor.
+
+        Args:
+            **kwargs: Optional; single_channel (bool): if True,
+                encoder will remove the last dimension of input if it is 1.
+                quality (int): The image quality, on a scale from 1 (worst) to 95 (best). Default: 95.
+                    For large images set quality <= 85.
+
+        Raises:
+            ValueError: If keyword arguments contain not supported arguments.
+        """
+        super().__init__()
+        self.codec_id = "webp"
+        webp_args = {"single_channel": True, "quality": 95}
+
+        diff = set(kwargs.keys()) - set(webp_args.keys())
+        if diff:
+            raise ValueError("Invalid args:", tuple(diff))
+
+        self.single_channel = kwargs.get("single_channel", webp_args["single_channel"])
+        self.quality = kwargs.get("quality", webp_args["quality"])
+
+    @property
+    def __name__(self):
+        return "webp"
+
+    def encode_single_image(self, image: np.ndarray) -> bytes:
+        """
+        Encode single image.
+
+        Example:
+            img = np.ones(100, 100, 3)
+            img_encoded = png_codec.encode_single_image(img)
+
+        Args:
+            image (np.ndarray): Single image to be encoded
+
+        Returns:
+            Encoded data.
+        """
+        with BytesIO() as buffer:
```

Comment:
are we able to have a persistent BytesIO object? instead of creating one every time a sample is encoded?

Reply:
Have tried this before, but anyway need to close it after each encoding and create in decodings, so not sure if it should be persistent.